src/odata-to-abstract-sql.ts

```diff
@@ -0,0 +1,1344 @@
+					break;
+				case 'POST':
+					tree = query.compile('InsertQuery');
+					break;
+				case 'DELETE':
+					tree = query.compile('DeleteQuery');
+					break;
+				default:
+					throw new SyntaxError(`Unknown method "${method}"`);
+			}
+		}
+		return {
+			tree,
+			extraBodyVars: this.extraBodyVars,
+			extraBindVars: this.extraBindVars,
+		};
+	}
+	PathSegment(method: string, bodyKeys: string[], path: any): Query {
+		if (!path.resource) {
+			throw new SyntaxError('Path segment must contain a resource');
+		}
+		const resource = this.Resource(path.resource, this.defaultResource);
+		this.defaultResource = resource;
+		const query = new Query();
+		// For non-GETs we bypass definitions for the actual update/insert as we need to write to the base table
+		const bypassDefinition = method !== 'GET';
+		query.fromResource(this.clientModel, resource, this, bypassDefinition);
+
+		// We can't use the ReferencedField rule as resource.idField is the model name (using spaces),
+		// not the resource name (with underscores), meaning that the attempts to map fail for a custom id field with spaces.
+		const referencedIdField: ReferencedFieldNode = [
+			'ReferencedField',
+			resource.tableAlias!,
+			resource.idField,
+		];
+		this.PathKey(method, path, query, resource, referencedIdField, bodyKeys);
+
+		if (path.options && path.options.$expand) {
+			this.Expands(resource, query, path.options.$expand.properties);
+		}
+		let bindVars: ReturnType<OData2AbstractSQL['BindVars']> | undefined;
+		if (path.property) {
+			const childQuery = this.PathSegment(method, bodyKeys, path.property);
+			query.merge(childQuery);
+			if (!path.property.resource) {
+				throw new SyntaxError('PathSegment has a property without a resource?');
+			}
+			const navigation = this.NavigateResources(
+				resource,
+				path.property.resource,
+			);
+			query.where.push(navigation.where);
+		} else if (path.link) {
+			if (!path.link.resource) {
+				throw new SyntaxError('PathSegment has a link without a resource?');
+			}
+			const linkResource = this.Resource(path.link.resource, resource);
+			let aliasedField: AliasNode<ReferencedFieldNode>;
+			let referencedField: ReferencedFieldNode;
+			if (this.FieldContainedIn(linkResource.resourceName, resource)) {
+				referencedField = this.ReferencedField(
+					resource,
+					linkResource.resourceName,
+				);
+				aliasedField = ['Alias', referencedField, linkResource.resourceName];
+			} else if (this.FieldContainedIn(resource.resourceName, linkResource)) {
+				referencedField = this.ReferencedField(
+					linkResource,
+					resource.resourceName,
+				);
+				aliasedField = ['Alias', referencedField, resource.resourceName];
+			} else {
+				throw new Error('Cannot navigate links');
+			}
+			this.PathKey(
+				method,
+				path.link,
+				query,
+				linkResource,
+				referencedField,
+				bodyKeys,
+			);
+			query.select.push(aliasedField);
+		} else if (
+			method == 'PUT' ||
+			method == 'PUT-INSERT' ||
```

Comment:
Because it's not a supported method to be passed in, it comes from https://github.com/balena-io-modules/odata-to-abstract-sql/pull/32/files#diff-3c71cc0b38f3742d3924bce55d96cf56R308 :
```ts
const insertQuery = this.PathSegment('PUT-INSERT', bodyKeys, path);
```
and is just an internal detail due to a `PUT` becoming two separate queries (INSERT + UPDATE) and needing to be handled slightly differently in the two cases